storage/ndb/rest-server/rest-api-server/internal/handlers/batchfeaturestore/handler.go

```
@@ -0,0 +1,204 @@
+			}
+			var fsError = fshandler.TranslateRonDbError(http.StatusBadRequest, ronDbErr.Error())
+			return fsError.GetStatus(), fsError.GetError()
+		}
+		var dbResponseIntf = getPkReadResponseJSON(numPassed, *metadata)
+		code, ronDbErr := h.dbBatchReader.Execute(readParams, *dbResponseIntf)
+		if log.IsDebug() {
+			log.Debugf("Rondb response: code: %d, error: %s, body: %s", code, ronDbErr, (*dbResponseIntf).String())
+		}
+		if ronDbErr != nil {
+			var fsError = fshandler.TranslateRonDbError(code, ronDbErr.Error())
+			return fsError.GetStatus(), fsError.GetError()
+		}
+		features = getFeatureValuesMultipleEntries(dbResponseIntf, fsReq.Entries, metadata, &featureStatus)
+	} else {
+		var emptyFeatures = make([][]interface{}, len(*fsReq.Entries))
+		for i := range(emptyFeatures) {
+			emptyFeatures[i] = make([]interface{}, metadata.NumOfFeatures)
+		}
+		features = &emptyFeatures
+	}
+	fsResp.Status = featureStatus
+	fillPassedFeaturesMultipleEntries(features, fsReq.PassedFeatures, &metadata.PrefixFeaturesLookup, &metadata.FeatureIndexLookup, &featureStatus)
+	fsResp.Features = *features
+	if fsReq.MetadataRequest != nil {
+		fsResp.Metadata = *fshandler.GetFeatureMetadata(metadata, fsReq.MetadataRequest)
+	}
+	return http.StatusOK, nil
+}
+
+func getFeatureValuesMultipleEntries(batchResponse *api.BatchOpResponse, entries *[]*map[string]*json.RawMessage, featureView *feature_store.FeatureViewMetadata, batchStatus *[]api.FeatureStatus) *[][]interface{} {
+	jsonResponse := (*batchResponse).String()
+	rondbResp := api.BatchResponseJSON{}
+	json.Unmarshal([]byte(jsonResponse), &rondbResp)
+	ronDbBatchResult := make([][]*api.PKReadResponseWithCodeJSON, len(*batchStatus))
+	batchResult := make([][]interface{}, len(*batchStatus))
+	for _, response := range *rondbResp.Result {
+		splitOperationId := strings.Split(*response.Body.OperationID, SEQUENCE_SEPARATOR)
+		seqNum, _ := strconv.Atoi(splitOperationId[0])
```

do not swallow error. handle the the error